test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ControllerActionExecutorTests.cs

```diff
@@ -323,18 +323,15 @@ public class ControllerActionExecutorTests
             var actionParameters = new Dictionary<string, object> { { "i", inputParam1 }, { "s", inputParam2 } };
 
             var dynamicTaskMethod = new ReturnTaskAsDynamicValue(_controller.ReturnTaskAsDynamicValue);
-            var expectedException = string.Format(
-                CultureInfo.CurrentCulture,
-                "The method 'ReturnTaskAsDynamicValue' on type '{0}' returned a Task instance even though it is not an asynchronous method.",
-                typeof(TestController));
 
-            // Act & Assert
-            var ex = await Assert.ThrowsAsync<InvalidOperationException>(
-               () => ExecuteAction(
-                   dynamicTaskMethod,
-                   _controller,
-                   actionParameters));
-            Assert.Equal(expectedException, ex.Message);
+            // Act
+            var result = await (Task<int>)(await ExecuteAction(
```

Comment:
The test name needs to change. Leave a comment here explaining that this isn't really a *supported* scenario, it's just something we don't check for specifically.